azure-sink-connector/src/integration-test/java/io/aiven/kafka/connect/azure/AvroIntegrationTest.java

```diff
@@ -0,0 +1,357 @@
+        connectorConfig.put("format.output.fields", "value");
+        connectorConfig.put("format.output.type", "avro");
+        connectorConfig.put("file.compression.type", compression);
+        connectorConfig.put("avro.codec", avroCodec);
+        getConnectRunner().createConnector(connectorConfig);
+
+        final int recordCountPerPartition = 10;
+        produceRecords(recordCountPerPartition);
+
+        final List<String> expectedBlobs = Arrays.asList(getAvroBlobName(0, 0, compression),
+                getAvroBlobName(1, 0, compression), getAvroBlobName(2, 0, compression),
+                getAvroBlobName(3, 0, compression));
+        awaitAllBlobsWritten(expectedBlobs.size());
+        assertIterableEquals(expectedBlobs, testBlobAccessor.getBlobNames(azurePrefix));
+
+        final Map<String, List<GenericRecord>> blobContents = new HashMap<>();
+        for (final String blobName : expectedBlobs) {
+            final byte[] blobBytes = getBlobBytes(testBlobAccessor.readBytes(blobName), compression);
+            final GenericDatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
+            try (SeekableInput sin = new SeekableByteArrayInput(blobBytes)) {
+                final List<GenericRecord> items;
+                try (DataFileReader<GenericRecord> reader = new DataFileReader<>(sin, datumReader)) {
+                    items = new ArrayList<>();
+                    reader.forEach(items::add);
+                }
+                blobContents.put(blobName, items);
+            }
+        }
+
+        // Connect will add two extra fields to schema and enrich it with
+        // connect.version: 1
+        // connect.name: input_data
+        final Schema avroInputDataSchemaWithConnectExtra = new Schema.Parser()
+                .parse("{\"type\":\"record\",\"name\":\"input_data\","
+                        + "\"fields\":[{\"name\":\"name\",\"type\":\"string\"}],"
+                        + "\"connect.version\":1,\"connect.name\":\"input_data\"}");
+        int cnt = 0;
+        for (int i = 0; i < recordCountPerPartition; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final String blobName = getAvroBlobName(partition, 0, compression);
+                final GenericData.Record expectedRecord = new GenericData.Record(avroInputDataSchemaWithConnectExtra);
+                expectedRecord.put("name", new Utf8("user-" + cnt));
+                cnt += 1;
+
+                final GenericRecord actualRecord = blobContents.get(blobName).get(i);
+                assertEquals(expectedRecord, actualRecord);
+            }
+        }
+    }
+
+    /**
+     * When Avro schema changes a new Avro Container File must be produced to Azure. Avro Container File can have only
+     * records written with same schema.
+     *
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws IOException
+     */
```

Comment:
nit: javadocs are a bit excessive here. I do like the comments though, adding more context beyond just the method name.